src/repr/src/row/encoding.rs

```diff
@@ -802,7 +802,7 @@ impl<'a> From<Datum<'a>> for ProtoDatum {
             Datum::Timestamp(x) => DatumType::Timestamp(x.into_proto()),
             Datum::TimestampTz(x) => DatumType::TimestampTz(x.into_proto()),
             Datum::Interval(x) => DatumType::Interval(x.into_proto()),
-            Datum::Bytes(x) => DatumType::Bytes(x.to_vec()),
+            Datum::Bytes(x) => DatumType::Bytes(Bytes::copy_from_slice(x)),
```

Comment:
wait, doesn't this literally call `.to_vec()`? https://docs.rs/bytes/1.5.0/src/bytes/bytes.rs.html#212-214

Reply:
You're totally right, `Datum::Bytes` would also need to use `Bytes` to make this zero-copy... which seems possible to do?

Reply:
> wait, doesn't this literally call .to_vec()?

Yup, but `impl From<Vec<u8>> for Bytes` has a special case to reuse the Vec alloc if it can. I feel like `Bytes::copy_from_slice` better declares the intention (one copy/alloc) than relying on that special case happening to match if we were to instead call `x.to_vec().into()` here.

> You're totally right, Datum::Bytes would also need to use Bytes to make this zero-copy... which seems possible to do? Not sure if that change is worth it though

Yup, but this is the encode path, which stays at 1 alloc per datum. The decode path drops an alloc because we're no longer decoding the proto into a `Vec<u8>` just for the Deref to wrap it into a `Datum::Bytes(&[u8])`. I'm skeptical that `Datum::Bytes(Bytes)` would play as well with all the various columnar memory layout tricks that compute plays.